src/main/java/com/fvv/bookstore/controller/BookControllerImpl.java

```diff
@@ -76,18 +82,23 @@ public void removeBook(final Long id) throws ControllerException {
 	/**
 	 * {@inheritDoc}
 	 */
-	public Book findBook(Long id) throws ControllerException {
+	public Book findBook(Long id) throws BookNotFoundException, ControllerException {
 		try {
 			return this.bookDao.findBook(id);
-		} catch (DaoException | BookNotFoundException e) {
+		} catch (BookNotFoundException e) {
```

Comment:
you can simply remove this catch here since you're re-throwing an existent one!

Reply:
ok